www/guides/react.md

```diff
@@ -0,0 +1,371 @@
+```
+
+Since the React code is rendering into an element with the ID `root`, you'll need to add that to `index.html`:
+
+```diff
+  <body>
+-    <h1>Welcome to Snowpack!</h1>
++    <div id="root"></div>
+    <script type="module" src="/index.js"></script>
+```
+
+Image: screenshot of the project, which shows "HELLO REACT" on a white background
+
+You've just created your first React component in Snowpack!
+
+## Customize your project layout
+
+Since you'll be adding a bunch of new files, you probably don't want them crowding up your top-level root directly. Snowpack is flexible enough to support whatever project layout that you prefer. In this guide, you'll learn how to use a popular project pattern from the React community.
+
+In this pattern the `src` directory is for components and their associated files. Then a `public` directory is for static assets like images, fonts, icons, and sometimes CSS.
+
+```bash
+📁 src
+↳ index.jsx
+📁 public
+↳ index.css
+↳ index.html
+```
+
+Use your favorite visual editor to rearrange and rename, or run these commands in the terminal:
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+Once your new project structure is in place, you need to tell Snowpack about it. Every Snowpack project comes with a `snowpack.config.js` file for any configuration that you might need. Right now, you should see a mostly empty, bare-bones configuration file. Add this to the empty `mount` object:
+
+```diff
+  mount: {
+-    /* ... */
++    public: '/',
++    src: '/_dist_',
+  },
+```
+
+`mount` is part of the [Snowpack Configuration API](https://www.snowpack.dev/#configuration). It allows you to customize the file structure of your project. The key is the name of the directory and the value is where you'd like them in the final build. With this new configuration, Snowpack builds files in `public` like `public/index.css` directory into `index.css`. It builds files in `src` like `src/index.js` into `/_dist_/index.js`, so you'll need to change that path in your `index.html`:
+
+```diff
+    <h1>Welcome to Snowpack!</h1>
+    <div id="root"></div>
+-    <script type="module" src="/index.js"></>
++    <script type="module" src="/_dist_/index.js"></script>
+  </body>
+```
+
+If Snowpack is still running, exit the process, you need to restart Snowpack for configuration file changes. When you start up again, if it worked, it should look the same.
+
+Create a new file at `src/App.jsx` and paste the following code into this new file to create an `App` component:
+
+```jsx
+import React, {useState, useEffect} from 'react';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
```

Comment:
We don’t need `setCount` in the memo here (but also it’s not hurting anything)

Reply:
some linters get mad if you don't :/ agreed it's not very important but good to do the right thing here in case anyone has one turned on by default

Reply:
that's just copy paste from the React template so maybe change it there?

Reply:
> some linters get mad if you don't :/

Yeah I always disabled those because they’re wrong! 😛  Queue another discussion about React hooks. (I’m fine for keeping, because again here it doesn’t hurt anything)